contracts/CategoryContract.sol

```diff
@@ -154,7 +158,7 @@ contract CategoryContract {
   }
 
   modifier categoryMustExists(uint256 id) {
-    require(categories[id].id > 0, "This category don't exists");
+    require(exists(id), "This category don't exists");
```

Comment:
Aqui um pequeno erro de inglês. Os verbos em inglês possuem conjugação de acordo com a pessoa em questão.

E no caso sempre quando é um gênero neutro, deve-se usar o 'it'. Como category é do gênero neutro, a conjugação correta é com o 'it'.

A conjugação de cada pessoa para o verbo 'do' fica assim:

I (Eu) - Do
You (Você) - Do
He (Ele) - Does
She (Ela) - Does
It (Neutro/objeto) - Does
We (Nós) - Do
You (Vocês) - Do

Sempre que usar o it portanto deverá acompanhar com a conjugação 'does', e não 'do' como foi usado nesse caso. 

No caso o correto ficaria assim: "This category doesn't exist". 

E sem o plural no final tbm. 

Deu pra entender?

Reply:
Deu sim, André! Vlw pelo review